### ISSUE TITLE
Support merging results?

### ISSUE THREAD

**Issue by bostanio** [palkan/test-prof#130]
How complex would it be to support merging test results? We're running tests in parallel, its easy to configure each one to spit out separate files but it would be great to be able to merge results across the entire test suite into one report of flamechart.

**Comment by palkan** [comment]
This is one of the features from our [Roadmap](https://github.com/palkan/test-prof/projects/1).

It's kinda complex feature, 'cause it depends on others:
1) first, we need a way to dump reports into files (#113); the idea is to provide Formatters API
2) then we need to  add the merging functionality itself: given N reports (in some format) produce an aggregated report (again, in some format); the main question here is where to trigger this aggregation? Probably, we need a CLI tool: `test-prof generate factory-flame path/to/report1 path/to/report2` or smth like that.

The 1) seems clear (from the implementation point of view), the 2) is not as clear: we need to think about a good UX for that.

So, will be glad to hear your suggestions)

**Comment by bostanio** [comment]
Sounds right. #113 seems pretty straight forward. IMO you'd dump the results to json or yml. I'd look to how Ruby exports a very simple json object for coverage reports (which makes them super easy to merge)

2. I'd imagine a set of rake tests for the CLI. Supporting glob patterns would be simplest. I'd think of it as:
```bundle exec rake test_prof:generate --type factory-flame --path path/to/**/*.fprof.json```

**Comment by modosc** [comment]
is this in progress at all or would you accept a pr / help

**Comment by palkan** [comment]
The full-featured support is planned for the v2 release cycle (no roadmap publicly available yet), which is going to standardize the profilers and tools APIs.

So, for the current version (which is going to be released as v1 later this year) I do plan to introduce such massive changes.

On the other hand, if you have some specific ideas or request (e.g., adding JSON support for a particular profiler)—I would be glad to accept a PR 🙂.

**Comment by panozzaj** [comment]
FYI https://evilmartians.com/chronicles/test-prof-3-guided-and-automated-ruby-test-profiling says that there is merging support, so perhaps this issue can be closed?

**Comment by palkan** [comment]
Yeah, it was implemented in a separate gem as a proof-of-concept (though we used it many times in "production" 😁).

This issue tracks the planned changes for v2, so let's keep it (even though we don't have a date yet).
